vertx-pg-client/src/test/java/io/vertx/pgclient/PgPoolTestBase.java

```diff
@@ -190,4 +204,26 @@ public void testCancelRequest(TestContext ctx) {
       ((PgConnection)conn).cancelRequest(ctx.asyncAssertSuccess());
     }));
   }
+
+  @Repeat(500)
+  @Test
+  public void checkBorderConditionBetweenIdleAndGetConnection(TestContext ctx) {
+    final int concurrentRequestAmount = 100;
+    final int idle = 1000;
+    Async async = ctx.async(concurrentRequestAmount);
+
+    options.setIdleTimeout(idle).setIdleTimeoutUnit(TimeUnit.MILLISECONDS);
+    poolOptions.setMaxSize(5).setIdleTimeout(idle).setIdleTimeUnit(TimeUnit.MILLISECONDS);
+
+    PgPool pool = createPool(new PgConnectOptions(options), poolOptions);
+
+    System.out.println("###################################################: ");
+    IntStream.range(0, concurrentRequestAmount).forEach(n -> CompletableFuture.runAsync(() ->
+      pool.query("SELECT CURRENT_TIMESTAMP;").execute(ctx.asyncAssertSuccess(rowSet -> {
+        for(Row row: rowSet){
+          async.countDown();
+          System.out.println("Connection #" + async.count() + " " + row.getOffsetDateTime(0));
+        }
+      }))));
+  }
```

Comment:
After the concurrent query test, we should also make sure no connection was leaked by sending a query to Postgres and inspecting the number of live connections.

Reply:
Do you mean something like this isn't?. If we are not reusing connections, then `pg_stat_activity'` will grow and grow...exceeding the poolSize.

Reply:
Right, but it is not explicit.